fallback.c

```diff
@@ -612,18 +612,45 @@ add_to_boot_list(CHAR16 *dirname, CHAR16 *filename, CHAR16 *label, CHAR16 *argum
 	if (EFI_ERROR(efi_status)) {
 		add_boot_option(dp, full_device_path, fullpath, label,
 				arguments);
-	} else if (option != 0) {
+	} else {
+		UINT16 bootnum;
 		CHAR16 *newbootorder;
-		newbootorder = AllocateZeroPool(sizeof (CHAR16) * nbootorder);
-		if (!newbootorder)
-			return EFI_OUT_OF_RESOURCES;
-
-		newbootorder[0] = bootorder[option];
-		CopyMem(newbootorder + 1, bootorder, sizeof (CHAR16) * option);
-		CopyMem(newbootorder + option + 1, bootorder + option + 1,
-			sizeof (CHAR16) * (nbootorder - option - 1));
-		FreePool(bootorder);
-		bootorder = newbootorder;
+		/* Search for the option in the current bootorder */
+		for (bootnum = 0; bootnum < nbootorder; bootnum++)
+			if (bootorder[bootnum] == option)
+				break;
+		if (bootnum == nbootorder) {
+			/* Option not found, prepend option and copy the rest */
+			newbootorder = AllocateZeroPool(sizeof (CHAR16)
```

Comment:
No space between `sizeof` and the open parenthesis (here and in the cases below).

Reply:
OK, I personally prefer a space between `sizeof` and type because it's a compiler directive, not a function call, but well, depends on the style I guess ...